pkg/client/delegation/client_test.go

```diff
@@ -0,0 +1,100 @@
+package delegation_test
+
+import (
+	"context"
+	"encoding/json"
+	"testing"
+	"time"
+
+	"cosmossdk.io/depinject"
+	"github.com/stretchr/testify/require"
+
+	"github.com/pokt-network/poktroll/pkg/client"
+	"github.com/pokt-network/poktroll/pkg/client/delegation"
+	"github.com/pokt-network/poktroll/testutil/sample"
+	"github.com/pokt-network/poktroll/testutil/testclient"
+	"github.com/pokt-network/poktroll/testutil/testclient/testeventsquery"
+	apptypes "github.com/pokt-network/poktroll/x/application/types"
+)
+
+const (
+	testTimeoutDuration = 100 * time.Millisecond
+
+	// duplicates pkg/client/delegation/client.go's delegationEventQuery for testing purposes.
+	delegationEventQuery = "message.action='pocket.application.EventDelegateeChange'"
+)
+
+func TestDelegationClient(t *testing.T) {
+	var (
+		expectedAddress         = sample.AccAddress()
+		expectedDelegationEvent = apptypes.EventDelegateeChange{
+			AppAddress: expectedAddress,
+		}
+		ctx = context.Background()
+	)
+
+	expectedEventBz, err := json.Marshal(expectedDelegationEvent)
+	require.NoError(t, err)
+
+	eventsQueryClient := testeventsquery.NewAnyTimesEventsBytesEventsQueryClient(
+		ctx, t,
+		delegationEventQuery,
+		expectedEventBz,
+	)
+
+	deps := depinject.Supply(eventsQueryClient)
+
+	// Set up delegation client.
+	delegationClient, err := delegation.NewDelegationClient(ctx, deps, testclient.CometLocalWebsocketURL)
```

Comment:
NIT:
```suggestion
	// NB: the URL passed to  `NewDelegationClient` is irrelevant here because `eventsQueryClient` is a mock.
	delegationClient, err := delegation.NewDelegationClient(ctx, deps, testclient.CometLocalWebsocketURL)
```